Lint: no-inner-declarations will warn std & cli/tests

The lint rule `no-inner-declarations` will come soon https://github.com/denoland/deno_lint/pull/297
When executing lint with this rule, we get lots of errors such as:

```
(no-inner-declarations) Move function declaration to function root
    function DateOverride(): void {
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    at /home/yusuktan/Repos/github.com/magurotuna/deno/cli/tests/unit/timers_test.ts:382:4

(no-inner-declarations) Move function declaration to module root
      async function asyncFn(): Promise<typeof value> {
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    at /home/yusuktan/Repos/github.com/magurotuna/deno/std/node/_util/_util_callbackify_test.ts:84:6

(no-inner-declarations) Move function declaration to module root
      function promiseFn(): Promise<typeof value> {
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    at /home/yusuktan/Repos/github.com/magurotuna/deno/std/node/_util/_util_callbackify_test.ts:96:6

(no-inner-declarations) Move function declaration to module root
      function thenableFn(): PromiseLike<any> {
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    at /home/yusuktan/Repos/github.com/magurotuna/deno/std/node/_util/_util_callbackify_test.ts:109:6

(no-inner-declarations) Move function declaration to module root
      async function asyncFn(): Promise<never> {
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    at /home/yusuktan/Repos/github.com/magurotuna/deno/std/node/_util/_util_callbackify_test.ts:140:6

(no-inner-declarations) Move function declaration to module root
      function promiseFn(): Promise<never> {
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    at /home/yusuktan/Repos/github.com/magurotuna/deno/std/node/_util/_util_callbackify_test.ts:169:6

(no-inner-declarations) Move function declaration to module root
      function thenableFn(): PromiseLike<never> {
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    at /home/yusuktan/Repos/github.com/magurotuna/deno/std/node/_util/_util_callbackify_test.ts:205:6

(no-inner-declarations) Move function declaration to module root
    async function asyncFn(arg: typeof value): Promise<typeof value> {
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    at /home/yusuktan/Repos/github.com/magurotuna/deno/std/node/_util/_util_callbackify_test.ts:249:4

(no-inner-declarations) Move function declaration to module root
    function promiseFn<T>(arg: typeof value): Promise<typeof value> {
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    at /home/yusuktan/Repos/github.com/magurotuna/deno/std/node/_util/_util_callbackify_test.ts:266:4

(no-inner-declarations) Move variable declaration to module root
              var type = FILETYPE_REGULAR_FILE;
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    at /home/yusuktan/Repos/github.com/magurotuna/deno/std/wasi/snapshot_preview1.ts:830:14

(no-inner-declarations) Move variable declaration to module root
              var type = FILETYPE_REGULAR_FILE;
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    at /home/yusuktan/Repos/github.com/magurotuna/deno/std/wasi/snapshot_preview1.ts:834:14

(no-inner-declarations) Move variable declaration to module root
              var type = FILETYPE_SYMBOLIC_LINK;
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    at /home/yusuktan/Repos/github.com/magurotuna/deno/std/wasi/snapshot_preview1.ts:838:14

(no-inner-declarations) Move variable declaration to module root
              var type = FILETYPE_REGULAR_FILE;
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    at /home/yusuktan/Repos/github.com/magurotuna/deno/std/wasi/snapshot_preview1.ts:842:14

Found 13 problems
```

We have to address these errors before the rule comes.

I want to make sure that how I will fix them is correct, so I have two questions:

1. Can the above `function` declarations that cause errors be replaced with arrow function? like `const asyncFn = async (): Promise<typeof value> => { }`
2. Can `var type = ..`  be rewritten from

https://github.com/denoland/deno/blob/fa65e49bc689b2b84a3fa2dd123fa616f430f7fe/std/wasi/snapshot_preview1.ts#L828-L846

to

```ts
let type: number;
switch (true) {
  case entries[i].isFile:
    type = FILETYPE_REGULAR_FILE;
    break;

  case entries[i].isDirectory:
    type = FILETYPE_REGULAR_FILE;
    break;

  case entries[i].isSymlink:
    type = FILETYPE_SYMBOLIC_LINK;
    break;

  default:
    type = FILETYPE_REGULAR_FILE;
    break;
}

entry_view.setUint8(20, type);
```

CC @caspervonb @bartlomieju

@bartlomieju 

> it seems strange `std_lint` shouldn't lint any code in `cli/tests/`

It's kind of weird... I ran `cargo test std_lint` command, but `cli/tests` files were included.

> did you check output from `eslint` with this rule?

I checked `eslint`, but I forgot enabling `no-inner-declarations` rule. my bad.
Checking again, then I've got output like:

```
/home/yusuktan/Repos/github.com/magurotuna/deno/cli/tests/unit/timers_test.ts
  382:5  error  Move function declaration to function body root  no-inner-declarations

/home/yusuktan/Repos/github.com/magurotuna/deno/std/node/_util/_util_callbackify_test.ts
   84:7  error  Move function declaration to function body root  no-inner-declarations
   96:7  error  Move function declaration to function body root  no-inner-declarations
  109:7  error  Move function declaration to function body root  no-inner-declarations
  140:7  error  Move function declaration to function body root  no-inner-declarations
  169:7  error  Move function declaration to function body root  no-inner-declarations
  205:7  error  Move function declaration to function body root  no-inner-declarations
  249:5  error  Move function declaration to function body root  no-inner-declarations
  266:5  error  Move function declaration to function body root  no-inner-declarations
```

--- a/std/wasi/snapshot_preview1.ts
+++ b/std/wasi/snapshot_preview1.ts
@@ -825,21 +825,22 @@ export default class Context {
           );
           entry_view.setUint32(16, name_data.byteLength, true);
 
+          let type: number;
           switch (true) {
             case entries[i].isFile:
-              var type = FILETYPE_REGULAR_FILE;
+              type = FILETYPE_REGULAR_FILE;
               break;
 
             case entries[i].isDirectory:
-              var type = FILETYPE_REGULAR_FILE;
+              type = FILETYPE_REGULAR_FILE;
               break;
 
             case entries[i].isSymlink:
-              var type = FILETYPE_SYMBOLIC_LINK;
+              type = FILETYPE_SYMBOLIC_LINK;
               break;
 
             default:
-              var type = FILETYPE_REGULAR_FILE;
+              type = FILETYPE_REGULAR_FILE;
               break;
           }
 

